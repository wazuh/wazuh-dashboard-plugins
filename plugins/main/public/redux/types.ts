import { DefaultRootState } from 'react-redux';

export type ResolverAction = {
  type: string;
  payload: any;
};

export type AppConfigState = {
  isLoading: boolean;
  isReady: boolean;
  hasError: boolean;
  data: { [key: string]: any };
};

export type AppRootState = DefaultRootState & {
  appConfig: AppConfigState,
};
