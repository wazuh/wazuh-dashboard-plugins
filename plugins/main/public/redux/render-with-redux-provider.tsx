import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react';
import storeRedux from './store';
import { Provider } from 'react-redux';
import type { RenderOptions } from '@testing-library/react';

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = storeRedux,
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
