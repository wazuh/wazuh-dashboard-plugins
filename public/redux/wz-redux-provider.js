/*
 * Wazuh app - React component for registering agents.
 * Copyright (C) 2015-2020 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 */
import React from 'react';
import { Provider } from 'react-redux';
import store from './store';

const WzReduxProvider = props => {
  return <Provider store={store}>{props.children}</Provider>;
};

export default WzReduxProvider;
