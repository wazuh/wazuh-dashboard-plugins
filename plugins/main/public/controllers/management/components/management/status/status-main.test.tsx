/*
 * Wazuh app - React test for Status component.
 *
 * Copyright (C) 2015-2022 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 *
 */

import React from 'react';
import WzStatus from './status-main';
import { renderWithProviders } from '../../../../../redux/render-with-redux-provider';

jest.mock('../../../../../kibana-services', () => ({
  getHttp: () => ({
    basePath: {
      prepend: str => str,
    },
  }),
  getUiSettings: () => ({
    get: (setting: string): any => {
      if (setting === 'theme:darkMode') {
        return false;
      }
    },
  }),
}));

describe('Status component', () => {
  it('renders correctly to match the snapshot', () => {
    const { container } = renderWithProviders(<WzStatus />);
    expect(container).toMatchSnapshot();
  });
});
