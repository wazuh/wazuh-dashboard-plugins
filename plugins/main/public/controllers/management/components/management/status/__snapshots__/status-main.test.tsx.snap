// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status component renders correctly to match the snapshot 1`] = `
<div>
  <div
    class="withUserLogged"
  >
    <img
      alt=""
      class="withUserLogged-logo"
      src="/plugins/wazuh/assets/images/themes/light/logo.svg"
    />
    <div
      class="euiSpacer euiSpacer--s"
    />
    <div
      class="euiText euiText--medium subdued-color"
    >
      Loading ...
    </div>
    <div
      class="euiSpacer euiSpacer--s"
    />
    <div
      class="euiProgress euiProgress--indeterminate euiProgress--xs euiProgress--primary withUserLogged-loader"
    />
  </div>
</div>
`;
